prow/tide/tide.go

```diff
@@ -919,9 +919,7 @@ func (c *Controller) accumulateBatch(sp subpool) (successBatch []PullRequest, pe
 		context := pj.Spec.Context
 		jobState := toSimpleState(pj.Status.State)
 		// Store the best result for this ref+context.
-		if s, ok := states[ref].jobStates[context]; !ok || s == failureState || jobState == successState {
```

Comment:
The implementation of `getBetterSimpleState` basically matches what is here. Funnily, the two inline checks that did this previously did it in a different ways (but both correct).